File: tree_practise/same_tree.py
from typing import Optional

class TreeNode:
    def __init__(self, val = 0, left = None , right = None):
        self.val = val 
        self.left = left
        self.right = right


    
class Solution :
    def sameTree (self, tree1 : Optional[TreeNode], tree2 : Optional[TreeNode]):
        
        if not p and not q :
            return True 
        
        if not p or not q :
            return False
        
        if p.val != q.val:
            return False 
        else:
            return __fn(p.right, q.right) and __fn(p.left, q.left)
            
    

    

File: tree_practise/sameTree.js
function TreeNode (val, left, right){
    this.val = (val === undefined ? 0 : val)
    this.left = left === undefined ? null : left
    this.right = right === undefined ? null : right

}


// @param {TreeNode} tree1
// @parm {TreeNode} tree2
// @return Boolean 

function isSameTree (p, q){
    if (!p && !q){
        return true 
    }
    if (!p || !q ){
        return false 
    }

    if (p.val != q.val){
        return False 
    }

    return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
}
